simulators/portal-interop/src/main.rs

```diff
@@ -587,6 +599,65 @@ dyn_async! {
     }
 }
 
+dyn_async! {
+    // test that a node will return a header via RECURSIVEFINDCONTENT that is stored locally and makes sure utp_transfer is false
+    async fn test_recursive_find_content_header<'a> (client_a: Client, client_b: Client) {
+        let header_with_proof_key: HistoryContentKey = serde_json::from_value(json!(HEADER_WITH_PROOF_KEY)).unwrap();
+        let header_with_proof_value: HistoryContentValue = serde_json::from_value(json!(HEADER_WITH_PROOF_VALUE)).unwrap();
+
+        match client_b.rpc.store(header_with_proof_key.clone(), header_with_proof_value.clone()).await {
+            Ok(result) => if !result {
+                panic!("Unable to store header with proof for recursive find content");
+            },
+            Err(err) => {
+                panic!("Error storing header with proof for recursive find content: {err:?}");
+            }
+        }
+
+        let target_enr = match client_b.rpc.node_info().await {
+            Ok(node_info) => node_info.enr,
+            Err(err) => {
+                panic!("Error getting node info: {err:?}");
+            }
+        };
+
+        match HistoryNetworkApiClient::add_enr(&client_a.rpc, target_enr.clone()).await {
```

Comment:
Why we are adding manually the enr into the routing table? Shouldn't `ping` do this automatically for us?

Reply:
resolved